### PR TITLE
Crud api calls

### PR DESCRIPTION
**Pull request by AaronGeesink**
Implemented the api endpoints for the DELETE, REPLACE, and CREATE operations. Right now, I can only be certain that the DELETE api call works properly. I will test the REPLACE and CREATE api calls more extensively after their function has been created in the operations layer.

This merge also contains Garrett's implementation of the Delete function in the operations layer. I merged his work into thsi branch so I could make double sure the OperationsLayer was working as intended.

NOTE: in the test runner, we create an OperationsLayer that is separate from the one created by the server OperationsLayer. This is only an issue for running unit tests, but shouldn't be an issue when running unit tests. I got around this by mimicing delete operations between the two operations layers when doing API calls to that they believe they match up.

### CODE REVIEW

**Comment by AaronGeesink** [comment]
This merge also contains Garrett's implementation of the Delete function in the operations layer.

File: server/routes/api.js [hunk 1]
```diff
@@ -24,4 +24,52 @@ router.post('/getFilteredData', (req, res) => {
...
+req: 
+    1) The id of the neighborhood to update within the neighborhoodList.
+    2) The column number to update (a number from 0 to 13)
+    3) The value to replace it with
+res: string message saying whether or not delete operation was successful.
+*/
+router.patch('/cards', (req, res) => {
+    console.log(req.body)
+    // result var store operation status. 0 means success, -1 means failure
+    var result = OperationsLayer.update(req.body[0], req.body[1], req.body[2])
```
Comment (raymondwzeng) [inline server/routes/api.js:68]: Will need to double-check this - I don't think numerical indexing will work. Though we're still not sure on how exactly to handle this case.

File: server/test/operationsTest.js [hunk 2]
```diff
@@ -3,16 +3,37 @@ const { expect } = require('chai');
...
         */
-        expect(operationsNeighborhoodList[8]["median_age"]).to.be.equal('42');
+        expect(OperationsLayer.getNeighborhoodList()[8]["median_age"]).to.be.equal('42');
+    })
+});
+
+describe("operations.deleteNeighborhood function", () => {
+	//const operationsNeighborhoodList = new OperationsLayer();
+    
+    //OperationsLayer.deleteNeighborhood(10); // Delete the neighborhood with an id of 10
```
Comment (raymondwzeng) [inline server/test/operationsTest.js:23]: Probably wanna remove leftover comment code if it won't be used elsewhere.

File: server/test/apiTest.js [hunk 3]
```diff
@@ -8,6 +8,41 @@ const analytics = require('../analytics');
 chai.use(require('chai-json'))
 chai.use(chaiHttp)
 
+describe("api/cards using router.delete", () => {
+    it("Should return successful delete message after deleting", () => {
+        //TODO: Add additional filters to the req once those filters are added to /api/getFilteredData
```
Comment (raymondwzeng) [inline server/test/apiTest.js:13]: If these are working, let's remove the TODOs.

**Review by raymondwzeng — COMMENTED** [review]
Looks good to me overall! Just a bit of cleanup if time permits, and maybe double check that update function.

File: server/routes/api.js (hunk 1 above)
Comment (AaronGeesink) [reply server/routes/api.js:68]: I expect to change this in integration. It'll be easier to test once I actually have a function to test it with.

**Pull request merged**